### PR TITLE
Drop jQuery for good

### PR DESCRIPTION
**Pull request by bfred-it**
I saw we were loading jQuery just for delegated events, `.has`, `.wrap` and minor invisible loops (e.g. `.addClass` instead of `for-of classList.add`)

Dropping 70KB is worth the few extra lines.

~~(untested proposal)~~ **Edit:** tested and fixed every part, seems to work, except the 2 pre-existing bugs below.

---

Aside: I wish JS had a few nice features to clean it up common operations:

- [optional chaining](https://github.com/tc39/proposal-optional-chaining) (avoids the many one-operation `if`s)
- [invoke on iterables](https://twitter.com/bfred_it/status/936550547556929538): e.g. `elements[].remove()` instead of `elements.forEach(el => el.remove())`

### CODE REVIEW

**Comment by bfred-it** [comment]
Mark as unread will need more testing and fixes, I think. Some issues are pre-existing, others may be new.

File: src/libs/smart-delegate.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	// Selector as base
+	if (typeof baseElement === 'string') {
+		baseElement = select.all(baseElement);
+	}
+
+	const [, ...argsWithoutBase] = args;
+	return baseElement.map(
+		element => delegate(element, ...argsWithoutBase)
+	);
+}
```
Comment (bfred-it) [inline src/libs/smart-delegate.js]: @sindresorhus I'm sure you'd prefer this outta hea, so https://github.com/zenorocha/delegate/pull/18 (but who knows if/when that happens)

Comment (sindresorhus) [reply src/libs/smart-delegate.js]: Yes, and if it doesn't get any activity, you can just create a better module for this.

Comment (bfred-it) [reply src/libs/smart-delegate.js]: Let's review and merge this in the meanwhile since more PRs might be created with jQuery.

Comment (bfred-it) [reply src/libs/smart-delegate.js]: Done ;)

**Comment by sindresorhus** [comment]
> invoke on iterables: e.g. elements[].remove() instead of elements.forEach(el => el.remove())

This could actually be implemented using `Proxy`. If a method is called on a decorated array and the method doesn't exist, call it on all the elements.

**Comment by sindresorhus** [comment]
Nice! 🙌

**Comment by bfred-it** [comment]
> This could actually be implemented using `Proxy`

How about `.classList.add('class')`? I'm not sure that'd work.

lodash' `invoke` feels more flexible, even though it's string-based:

```js
_.invoke(elements, 'remove');
_.invoke(elements, 'classList.add', 'class');
```

**Comment by sindresorhus** [comment]
@bfred-it That would work too. The Proxy could just wrap all nested properties in the Proxy too.

**Comment by bfred-it** [comment]
Looking forward to your module for `all(elements).classList.add('class')` 😄

**Pull request merged**